packages/core/src/repository/repository.service.ts

```diff
@@ -9,8 +11,44 @@ export type Repository = {
     fullName: string; // owner/name format
 }
 
+@injectable()
 export class RepositoryService {
+    constructor(
+        @multiInject(REPOSITORY_ADAPTER_IDENTIFIER) @optional() private adapters: RepositoryAdapter[] = []
+    ) {}
+
     async getRepository(): Promise<Repository> {
+        // Try to auto-detect using adapters first
+        const detectedAdapter = await this.autoDetectAdapter();
+        if (detectedAdapter) {
+            return await detectedAdapter.getRepository();
+        }
+
+        // Fallback to basic implementation
+        return this.getBasicRepository();
+    }
+
+    /**
+     * Auto-detect the appropriate repository adapter for the current context
+     */
+    private async autoDetectAdapter(): Promise<RepositoryAdapter | null> {
+        for (const adapter of this.adapters) {
+            try {
+                if (await adapter.supports()) {
+                    return adapter;
+                }
+            } catch (error) {
+                // Continue to next adapter if this one fails
+                console.warn(`Repository adapter failed to check support: ${String(error)}`);
+            }
```

Comment:
No try catch, it should not occurred some errors, if it fails it is an unexpected error so it will fail at higher level